AIP-7726 add default cards

talebzeghmi opened the pull request:
The reason is to add default cards in WFSDK as part of the push,
* see example 
   * https://argo-server.int.stage-k8s.zg-aip.net/argo-ui/archived-workflows/aip-playground-dev/447be931-ca09-470c-a6e4-b23a76ca9baf
   * click on default_card.html, it'll be blank but then click on on download DOWNLOAD_CARD.HTML and inspect source to see the HTML and error.  You could download the HTML then load it locally and the Metaflow card would correctly render.
   * https://metaflow.int.stage-k8s.zg-aip.net/NBFlow/argo-nbflow-svhd9/start/start-1?group=true&order=startTime&direction=asc

However we need to update ARGO_ARTIFACT_CONTENT_SECURITY_POLICY
* https://github.com/argoproj/argo-workflows/issues/9121
* https://github.com/argoproj/argo-workflows/blob/master/server/artifacts/artifact_server.go#L462-L464

talebzeghmi committed:
AIP-7726 add default cards

talebzeghmi committed:
param fix

cloudw commented on metaflow/plugins/aip/aip.py:
```diff
@@ -286,8 +286,23 @@ def _create_workflow_yaml(
 
+    @staticmethod
+    def _add_archive_section_to_cards_artifacts(workflow: dict):
+        print("in _add_archive_section_to_cards_artifacts")
```
Just a reminder to remove all debugging print, or update them to make more sense to users.

cloudw commented on metaflow/plugins/aip/aip.py:
```diff
@@ -286,8 +286,23 @@ def _create_workflow_yaml(
         for key, value in self._get_flow_labels().items():
             workflow["metadata"]["labels"][key] = value
 
+        KubeflowPipelines._add_archive_section_to_cards_artifacts(workflow)
```
What does archive section do?

cloudw commented on metaflow/plugins/aip/aip_metaflow_step.py:
```diff
@@ -398,6 +427,28 @@ def aip_metaflow_step(
+        script_name,
+    )
+    cmd: str = card_cli.format(run_id=metaflow_run_id)
+    cmd = f"{cmd} > /tmp/outputs/cards/default_card.html"
```
Does this mean that only default card is supported, but user can't add any other card yet?

talebzeghmi replied:
added comments no for it now, thanks!

talebzeghmi replied:
added an example to the repo of multiple card 

see:

https://metaflow.int.stage-k8s.zg-aip.net/NBFlow/argo-nbflow-lbnlj/end/end-3?direction=asc&group=true&order=startTime&section

cloudw commented on metaflow/plugins/aip/aip_metaflow_step.py:
```diff
@@ -398,6 +427,28 @@ def aip_metaflow_step(
+
+    if process.returncode != 0:
+        logging.info(f"---- Following command returned: {process.returncode}")
+        logging.info(cmd.replace(" && ", "\n"))
```
Maybe `cmd.replace(" && ", "\n&& ")` so that you get new lines without losing the command chaining semantics.

talebzeghmi replied:
this is a nice to have, but skipped it for this change, as it already exists and is for debugging only

thank you!

talebzeghmi committed:
PR feedback
- comment archive none
- add card id to support current.card.append()
- add example of notebook card & multiple cards

talebzeghmi committed:
aip.zillowgroup.net/aip-pod-default -> aip.zillowgroup.net/aip-wfsdk-pod
